### PR TITLE
Fixes #15340 - Org destroy unassociates hostgroups

### PR DESCRIPTION
**Pull request by parthaa**
Org destroy was not properly unassociating environments and content
views from hosts + hostgroups.

Updated the plan to handle that

### CODE REVIEW

**Comment by jlsherrill** [comment]
you gotta delete the facets!

**Comment by parthaa** [comment]
@jlsherrill updated

File: app/views/overrides/taxonomies/_action_buttons.html.erb [hunk 1]
```diff
@@ -12,7 +12,7 @@
     display_link_if_authorized(_("Edit"), hash_for_edit_taxonomy_path(taxonomy) ),
     display_link_if_authorized(_("Nest"), hash_for_nest_taxonomy_path(taxonomy) ),
     display_link_if_authorized(_("Clone"), hash_for_clone_taxonomy_path(taxonomy) ),
-    display_delete_if_authorized(hash_for_taxonomy_path(taxonomy), :confirm => _("Delete %s?") % taxonomy.name, :action => :destroy),
+    display_delete_if_authorized(hash_for_taxonomy_path(taxonomy), :data => { :confirm => _("%s %s has %s Hosts and %s Hostgroups that will need to be reassociated post deletion. Delete %s?") % [taxonomy_title, taxonomy.name, taxonomy.hosts.count, taxonomy.hostgroups.count, taxonomy.name]}, :action => :destroy),
```
Comment (jlsherrill) [inline app/views/overrides/taxonomies/_action_buttons.html.erb]: should these be taxonomy.unscoped.hosts (same for hostgroups).  I'm seeing 0 unless i switch to any context

**Comment by jlsherrill** [comment]
Seeing an issue during org destroy:

1) register a host
2) subscribe it to some subscription
3) delete the organization

results:  

Task becomes paused stuck on the finalize phase of:  Actions::Katello::Host::Destroy

RestClient::ResourceNotFound

when trying to reindex the pool which has been deleted

**Comment by jlsherrill** [comment]
Rebase please

**Comment by parthaa** [comment]
[test]

**Comment by parthaa** [comment]
[test]

**Comment by zjhuntin** [comment]
When trying out with PR I was not able to delete the organization with a host & subscription. The error I get is: 

> Cannot delete record because of dependent content_facets

**Comment by ehelms** [comment]
@parthaa planning to revisit this PR?

**Comment by parthaa** [comment]
I believe https://github.com/Katello/katello/pull/6615 will fix the issue the @zjhuntin hit w.r.t. Going to wait that PR goes in.

**Comment by parthaa** [comment]
@zjhuntin @jlsherrill can either of you re review...

**Comment by zjhuntin** [comment]
I managed to delete my test organization after the most recent update! 🎆  APJ

**Pull request merged**